flash/data/batch.py

```diff
@@ -0,0 +1,138 @@
+from typing import Any, Callable, Mapping, Optional, Sequence, Union
+
+import torch
+from pytorch_lightning.trainer.states import RunningStage
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+
+from flash.data.utils import _contains_any_tensor, convert_to_modules
+
+
+class _Chainer(torch.nn.Module):
+
+    def __init__(
+        self,
+        per_sample_pre_tensor_transform: Callable,
+        per_sample_to_tensor_transform: Callable,
+        per_sample_post_tensor_transform: Callable,
+        assert_contains_tensor: bool = False
+    ):
+        super().__init__()
+
+        self.per_sample_pre_tensor_transform = convert_to_modules(per_sample_pre_tensor_transform)
+        self.per_sample_to_tensor_transform = convert_to_modules(per_sample_to_tensor_transform)
+        self.per_sample_post_tensor_transform = convert_to_modules(per_sample_post_tensor_transform)
+        self.assert_contains_tensor = assert_contains_tensor
+
+    def forward(self, sample: Any):
+        sample = self.per_sample_pre_tensor_transform(sample)
+        sample = self.per_sample_to_tensor_transform(sample)
```

Comment:
add some comments why is first called `pre` and subsequently `to`

Reply:
sure - I guess with more documentation will be easy to understand the process

Reply:
Yes, we will add more in a following PR. Right now, let's merge this one to unlock dev on Flash